### PR TITLE
Remove AzureServiceBusTransportConfigContext to simplify acceptance tests

### PR DESCRIPTION
**Pull request by danielmarbach**
- Remove AzureServiceBusTransportConfigContext to simplify acceptance setup
- Cleanup and align tests
- Introduced TestSuiteConstraints for ForwardingToplogy and EndpointOrientedToplogy and removed redundant feature code to detect topology

### CODE REVIEW

File: src/AcceptanceTests/Routing/When_using_single_namespace.cs [hunk 1]
```diff
@@ -63,8 +37,13 @@ public class SourceEndpoint : EndpointConfigurationBuilder
         {
             public SourceEndpoint()
             {
-                EndpointSetup<DefaultServer>(endpointConfiguration =>
-                        endpointConfiguration.ConfigureTransport().Routing().RouteToEndpoint(typeof(MyRequest), typeof(TargetEndpoint)));
+                EndpointSetup<DefaultServer>(c =>
+                {
+                    var transport = c.ConfigureAzureServiceBus();
```
Comment (yvesgoeleven) [inline src/AcceptanceTests/Routing/When_using_single_namespace.cs]: How did you prevent the automatic invocation of this method next to this explicit invocation ?

**Comment by danielmarbach** [comment]
Test that is failing is also failing on other branch https://builds.particular.net/viewLog.html?buildId=238150&tab=buildResultsDiv&buildTypeId=NServiceBusAzure_NServiceBusAzureServiceBus_3RunAcceptenceTestsWithForwardingTop

File: src/AcceptanceTests/Routing/When_using_single_namespace.cs (hunk 1 above)
Comment (yvesgoeleven) [reply src/AcceptanceTests/Routing/When_using_single_namespace.cs]: I see this just exposes our config, what I really meant is how are this config and ConfigureEndpointAzureServiceBusTransport not overwriting each other

Comment (danielmarbach) [reply src/AcceptanceTests/Routing/When_using_single_namespace.cs]: Not sure I follow @yvesgoeleven

**Comment by yvesgoeleven** [comment]
I think that test is checking the backward compat when there happens to exist more entities in the bundle, that entity 'bundle-2' was probably created manually. Should extend the test to create that entity if it does not exist (using a new instance of namespace manager)

File: src/AcceptanceTests/Routing/When_using_single_namespace.cs (hunk 1 above)
Comment (yvesgoeleven) [reply src/AcceptanceTests/Routing/When_using_single_namespace.cs]: Explained in zoom, it was more a question of me not understanding the order of invocation

File: src/AcceptanceTests/RequiresAdditionally.cs [hunk 2]
```diff
@@ -0,0 +1,23 @@
+﻿namespace NServiceBus.AcceptanceTests
+{
+    using NUnit.Framework;
+
+    public static class RequiresAdditionally
```
Comment (SeanFeldman) [inline src/AcceptanceTests/RequiresAdditionally.cs]: Nice!

File: src/AcceptanceTests/Routing/When_publishing_and_subscribing_to_self_with_overridden_address.cs [hunk 3]
```diff
@@ -30,9 +30,9 @@ public class PublisherAndSubscriber : EndpointConfigurationBuilder
         {
             public PublisherAndSubscriber()
             {
-                EndpointSetup<DefaultPublisher>(builder =>
+                EndpointSetup<DefaultPublisher>(c =>
```
Comment (SeanFeldman) [inline src/AcceptanceTests/Routing/When_publishing_and_subscribing_to_self_with_overridden_address.cs]: "c" as in "config" or "endpointConfig"? 😉

File: src/AcceptanceTests/Routing/When_scaling_out_senders_that_uses_callbacks.cs [hunk 4]
```diff
@@ -71,8 +71,8 @@ public class ScaledOutClient : EndpointConfigurationBuilder
         {
             public ScaledOutClient()
             {
-                EndpointSetup<DefaultServer>(endpointConfiguration =>
-                    endpointConfiguration.ConfigureTransport().Routing().RouteToEndpoint(typeof(MyRequest), typeof(ServerThatRespondsToCallbacks)));
+                EndpointSetup<DefaultServer>(c =>
```
Comment (SeanFeldman) [inline src/AcceptanceTests/Routing/When_scaling_out_senders_that_uses_callbacks.cs]: Personally, I see this as unnecessary change. The name was descriptive, and now it's "go figure". "c" as in "context"? Please revert.

File: src/AcceptanceTests/Routing/When_sending_to_specific_namespace.cs [hunk 5]
```diff
@@ -13,56 +11,38 @@ namespace NServiceBus.Azure.Transports.WindowsAzureServiceBus.AcceptanceTests.Ro
 
     public class When_using_multiple_namespaces : NServiceBusAcceptanceTest
     {
+        static string connectionString = connectionString = EnvironmentHelper.GetEnvironmentVariable("AzureServiceBusTransport.ConnectionString");
```
Comment (SeanFeldman) [inline src/AcceptanceTests/Routing/When_sending_to_specific_namespace.cs]: We sprinkle these these environment variable names like there's no tomorrow 😄 
While there's no issue for now, we'll have to consolidate it in another PR for issue #626.

File: src/AcceptanceTests/Routing/When_sending_to_specific_namespace.cs [hunk 6]
```diff
@@ -13,56 +11,38 @@ namespace NServiceBus.Azure.Transports.WindowsAzureServiceBus.AcceptanceTests.Ro
...
-                }
-            };
-
-            runSettings.Set("AzureServiceBus.AcceptanceTests.TransportConfigContext", ctx);
-
             var context = await Scenario.Define<Context>()
                 .WithEndpoint<EndpointInSourceNamespace>(b =>
                 {
-                    b.When(async (bus, c) =>
+                    b.CustomConfig(c =>
```
Comment (SeanFeldman) [inline src/AcceptanceTests/Routing/When_sending_to_specific_namespace.cs]: Much cleaner,  ❤️  it.
Giving credit to @yvesgoeleven for the original creativity though 😃

File: src/AcceptanceTests/Routing/When_sending_to_specific_namespace.cs [hunk 7]
```diff
@@ -126,48 +84,26 @@ public class When_using_multiple_namespaces : NServiceBusAcceptanceTest
...
+           var context = await Scenario.Define<Context>()
                 .WithEndpoint<EndpointInSourceNamespace>(b =>
                 {
-                    b.When((bus, c) => bus.Send(new MyRequest()));
+                    b.CustomConfig(c =>
+                    {
+                        var transport = c.ConfigureAzureServiceBus();
+                        transport.NamespacePartitioning().AddNamespace("source", connectionString);
+                        var targetNamespace = transport.NamespaceRouting().AddNamespace("target", targetConnectionString);
+                        targetNamespace.RegisteredEndpoints.Add(Conventions.EndpointNamingConvention(typeof(EndpointInTargetNamespace)));
```
Comment (SeanFeldman) [inline src/AcceptanceTests/Routing/When_sending_to_specific_namespace.cs]: I'd still name it `TestingConventions`. `Conventions` is overloaded (Core has it).

File: src/AcceptanceTests/Routing/When_subscribing_to_a_derived_polymorphic_event_with_forwarding_topology.cs [hunk 8]
```diff
@@ -4,18 +4,18 @@
...
     using Features;
-    using Settings;
     using NUnit.Framework;
 
     public class When_subscribing_to_base_and_derived_polymorphic_events_with_forwarding_topology : NServiceBusAcceptanceTest
     {
         [Test]
         public async Task Should_handle_each_event_once_only()
         {
+            RequiresAdditionally.ForwardingToplogy();
```
Comment (SeanFeldman) [inline src/AcceptanceTests/Routing/When_subscribing_to_a_derived_polymorphic_event_with_forwarding_topology.cs]: Now that I look at it, `RequiresAdditionally` is a bit misleading. It **requires** it and period. Only when it's the right topology, then the test can run. Rather see `Requires.ForwardingToplogy()`

Comment (SeanFeldman) [reply src/AcceptanceTests/Routing/When_subscribing_to_a_derived_polymorphic_event_with_forwarding_topology.cs]: Or `TestRequires.ForwardingToplogy()`

File: src/AcceptanceTests/Routing/When_subscribing_to_a_derived_polymorphic_event_with_forwarding_topology.cs [hunk 9]
```diff
@@ -61,59 +56,27 @@ public class Publisher : EndpointConfigurationBuilder
...
-            }
         }
 
 
         public class Subscriber : EndpointConfigurationBuilder
         {
             public Subscriber()
             {
-                EndpointSetup<DefaultServer>(endpointConfiguration =>
+                EndpointSetup<DefaultServer>(c =>
```
Comment (SeanFeldman) [inline src/AcceptanceTests/Routing/When_subscribing_to_a_derived_polymorphic_event_with_forwarding_topology.cs]: Negative, sir. Unnecessary change.

File: src/AcceptanceTests/Routing/When_subscribing_to_a_derived_polymorphic_event_with_forwarding_topology.cs [hunk 10]
```diff
@@ -61,59 +56,27 @@ public class Publisher : EndpointConfigurationBuilder
 
             public Publisher()
             {
-                EndpointSetup<DefaultPublisher>(endpointConfiguration =>
+                EndpointSetup<DefaultPublisher>(c =>
```
Comment (SeanFeldman) [inline src/AcceptanceTests/Routing/When_subscribing_to_a_derived_polymorphic_event_with_forwarding_topology.cs]: c? si senior.

File: src/AcceptanceTests/Routing/When_unsubscribing.cs [hunk 11]
```diff
@@ -51,11 +47,7 @@ public class Endpoint : EndpointConfigurationBuilder
         {
             public Endpoint()
             {
-                EndpointSetup<DefaultServer>(endpointConfiguration =>
-                {
-                    endpointConfiguration.EnableFeature<DetermineWhatTopologyIsUsed>();
-                    endpointConfiguration.ConfigureTransport().Routing().RouteToEndpoint(typeof(MyEvent), typeof(Endpoint));
-                });
+                EndpointSetup<DefaultServer>(c => c.ConfigureTransport().Routing().RouteToEndpoint(typeof(MyEvent), typeof(Endpoint)));
```
Comment (SeanFeldman) [inline src/AcceptanceTests/Routing/When_unsubscribing.cs]: c c c I know! https://www.youtube.com/watch?v=EzZvY8LZ7Gw

File: src/AcceptanceTests/Routing/When_subscribing_to_a_derived_polymorphic_event_with_forwarding_topology.cs (hunk 8 above)
Comment (danielmarbach) [reply src/AcceptanceTests/Routing/When_subscribing_to_a_derived_polymorphic_event_with_forwarding_topology.cs]: Requires is already taken. I can rename to TestRequires if you prefer

File: src/AcceptanceTests/Routing/When_sending_to_specific_namespace.cs (hunk 7 above)
Comment (danielmarbach) [reply src/AcceptanceTests/Routing/When_sending_to_specific_namespace.cs]: In the acceptance tests it is almost always the acceptance test conventions. I'm leaving it as is. Much cleaner that way

**Comment by SeanFeldman** [comment]
Convetions and RequiresAdditionally will be addressed. ATTs argument names will be addressed at maintainers sync.
